src/dtls13.c

```diff
@@ -1998,6 +1998,13 @@ int Dtls13NewEpoch(WOLFSSL* ssl, w64wrapper epochNumber, int side)
             return BAD_STATE_E;
     }
 
+#ifndef WOLFSSL_TLS13_IGNORE_AEAD_LIMITS
+    /* We are updating the receiving keys for this connection. We can restart
+     * the failed decryption counter. */
+    if (side == ENCRYPT_AND_DECRYPT_SIDE || side == DECRYPT_SIDE_ONLY)
+        w64Zero(&ssl->macDropCount);
```

Comment:
I think this will not be precise, we have multiple epochs and we can change epochs, but `macDropCount` is shared. I think it would be more correct to maintain a per epoch counter.

Reply:
Also, consider removing an epoch if, once we receive the ack of the key update, its macDrop counter is greater than our limit. We will not decrypt anymore with that epoch.

Reply:
I don't think we need a per epoch granularity. Adding marking old epochs as encrypt only so that they can not be used for decryption.

Reply:
But then you have also to mark old epochs as encrypt only every time you receive a new decrypting epoch, otherwise sending a new decrypting epoch can be used to reset the counter. I think adding a per-epoch counter may be a more explicit, clean, and easy-to-audit solution.
Is the memory usage concerning you? 
Another option would be not allowing multiple decryption epochs after the handshake, but it would be a major refactor and we still need mulitple epochs logic for the handshake.

Reply:
Added a per epoch counter. My main concern was not changing too much with this PR.